janusgraph_python/structure/io/graphsonV3d0.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2023 JanusGraph-Python Authors Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections import OrderedDict
+from gremlin_python.structure.io.graphsonV3d0 import _GraphSONTypeIO, EdgeDeserializer, GraphSONUtil, GraphSONReader, GraphSONWriter
+from gremlin_python.structure.graph import Edge, Vertex
+from janusgraph_python.process.traversal import _JanusGraphP, RelationIdentifier
+
+class JanusGraphSONReader(GraphSONReader):
+    def __init__(self):
+        # register JanusGraph-specific RelationIdentifier deserializer
+        deserializer_map = {
+            'janusgraph:RelationIdentifier': JanusGraphRelationIdentifier
+        }
+        GraphSONReader.__init__(self, deserializer_map)
+
+class JanusGraphSONWriter(GraphSONWriter):
+    def __init__(self):
+        # register JanusGraph-specific RelationIdentifier and text-predicate serializer
+        serializer_map = [
+            (RelationIdentifier, JanusGraphRelationIdentifier),
+            (_JanusGraphP, JanusGraphPSerializer)
+        ]
+        GraphSONWriter.__init__(self, serializer_map)
+
+
+class JanusGraphPSerializer(_GraphSONTypeIO):
+    @classmethod
+    def dictify(cls, p, writer):
+        out = {"predicate": p.operator,
+               "value": [writer.to_dict(p.value), writer.to_dict(p.other)] if p.other is not None else
+               writer.to_dict(p.value)}
+        return GraphSONUtil.typed_value("JanusGraphP", out, "janusgraph")
+
+class JanusGraphRelationIdentifier(_GraphSONTypeIO):
```

Comment:
```suggestion
class JanusGraphRelationIdentifierSerializer(_GraphSONTypeIO):
```

(Just to make it clear that this is the serializer and not the `RelationIdentifier` itself.

Reply:
Great catch, you are right that distinguish JanusGraphRelationIdentifier and RelationIdentifier would be great. 

Following the naming convention used in https://github.com/apache/tinkerpop/blob/master/gremlin-python/src/main/python/gremlin_python/structure/io/graphsonV3d0.py:
* if something only serializes, it is called 'XyzSerializer'
* if something only deserializes, it is called 'XyzDeserializer'
* if something both serializes and  deserializes, it is called 'XyzIO'

How about if we follow that convention and we call it JanusGraphRelationIdentifierIO?

Reply:
Yes sure, it definitely makes sense to follow that convention as JanusGraph-Python contributors probably also have some knowledge of the Gremlin-Python code.
The other option would be to split this into a serializer and deserializer, but that's your choice.

Reply:
Finally I decided to call it JanusGraphRelationIdentifierIO. Thanks.